### DISCUSSION TITLE
How to bundle node_modules code together with my code for one dynamic import?

### DISCUSSION THREAD

**Discussion by Daniel15** [webpack/webpack#14328] [Q&A]
Hi!

My app currently splits the app code and vendor (ie. node_modules) code into separate chunks. I think this is the default chunk config, and it's great.

However, now I want to lazy load one of my modules that uses vendor code. This particular third-party code is only used by one module in my app (a small wrapper around the third-party library), which I want to be lazy loaded. I used `import` like:
```js
import(/* webpackChunkName: "SomeModule" */ './SomeModule').then(...)
```
where `SomeModule` is a module in my own code. This works fine, but it's generating two chunks: A `SomeModule` chunk that's very small (<1 KB), and a chunk just identified by an ID (`176` in my case) that has the vendor code.

I guess this makes sense since the default config is splitting the `node_modules` from app code, but since the `SomeModule` module is the only module that imports the vendor code, I'd like to bundle both `SomeModule` **and** the vendor code into a single chunk, as it doesn't make sense to split them in this case (my code is just a small wrapper around the vendor code and is very unlikely to change, so there's no real caching benefits of splitting these).

How can I accomplish this? I think what I want to say is "For the `SomeModule` chunk, also bundle the node_modules into it", but I don't know how to do that. Currently, `splitChunks` is configured as `chunks: 'all'`, and I don't have any other chunk-related config in my Webpack config.

Thanks!

**Comment by nobodyguy** [comment]
Same question here:
I have a dynamically imported React component that uses OpenLayers functions. OpenLayers + other geo dependencies are rather big, so I'd like to include it in the chunk that is already generated for my imported React component.
So basically my Map component with all the geo dependencies will be lazy loaded only when a user navigates to that page.
How to achieve it?
